### ISSUE TITLE
Properly detecting order in components/routes/controllers/models with destructuring and new module imports

### ISSUE THREAD

**Issue by jbandura** [ember-cli/eslint-plugin-ember#110]
## Problem (demonstrated on services)
Currently we're only looking at type of function when deciding about the order. For example while this would be detected properly:
```
export default Ember.Component.extend({
  service: service(),
});
```
..this one would fail (even though it's perfectly valid):
```
const { foo } = Ember.inject.service();
export default Ember.Component.extend({
  service: foo()
});
```

## Solution
We need to keep track of destructured names for the given functions and compare the one we get is the one that was destructured (similiar work had to be done for the `no-global-jquery` rule).

**Comment by michalsnik** [comment]
I think it's something we should do alongside work on new module imports. Let's move this example there and keep all `new-modules` related stuff in one issue. What do you think @jbandura ?

**Comment by rwwagner90** [comment]
Since module imports will be enforced, and destructuring will be going away, is this needed?

```js
import {inject as service} from '@ember/service';
foo: service()
```

Works just fine, and I feel like we could use this time to work on other, more pertinent rules.

**Comment by michalsnik** [comment]
This works @rwwagner90 because we detect services as `inject` and `services` and if you use:
```
import {inject as somthing} from '@ember/service';
foo: something()
```
it won't. It's not a big deal though, who would ever want to override such thing with completely custom name, especially in ember where everything is well organised and opinionated. I'm ok with chilling with this a bit. We can improve this in few months and maybe even do a major release then without supporting global `Ember` module anymore. This way we'll be able to actually move forward and not introduce crazy logic behind every check.

**Comment by rwwagner90** [comment]
Yeah, seems silly to want to call `service` something other than `service`. Ember is opinionated for a reason.
